crates/stages/src/stages/bodies.rs

```diff
@@ -106,45 +106,22 @@ impl<DB: Database, D: BodyDownloader, C: Consensus> Stage<DB> for BodyStage<D, C
         // Get id for the first transaction in the block
         let mut first_tx_id = db.get_first_tx_id(starting_block)?;
 
-        // Cursor used to look up headers for block pre-validation
-        let mut header_cursor = db.cursor::<tables::Headers>()?;
-
         // NOTE(onbjerg): The stream needs to live here otherwise it will just create a new iterator
         // on every iteration of the while loop -_-
         let mut bodies_stream = self.downloader.bodies_stream(bodies_to_download.iter());
         let mut highest_block = previous_block;
-        while let Some((block_number, header_hash, body)) =
+        while let Some(block) =
             bodies_stream.try_next().await.map_err(|err| StageError::Internal(err.into()))?
```

Comment:
If the bodies stream returns an error (which it will given the fix recommended below), then we should not propagate the error up, because it will cause the pipeline to unwind the stage.

```
whilet let Some(block) = bodies_stream.next() {
      let block = match block {
            Ok(inner) => inner,
           Err(err => error!(err); break; // This should be `break` to avoid gaps in the blocks.
      }
      ...
}
```

Reply:
This generally feels like the pipeline has a massive footgun because the pipeline is doing `pipeline.run(..).await?` which will "close" the pipeline on any error. This makes me think that we should handle pipeline errors at the pipeline layer? https://github.com/paradigmxyz/reth/blob/ae2fc494c8d8d863f484ab53d92ce6a04bfdf1f4/crates/stages/src/pipeline.rs#L189

cc @onbjerg let's track on separate issue and figure out a good solution here

Reply:
the local issue is fixed, the pipeline stuff still needs to be handled